workflow/controller/operator.go

```diff
@@ -93,6 +96,7 @@ func newWorkflowOperationCtx(wf *wfv1.Workflow, wfc *WorkflowController) *wfOper
 		}),
 		controller:    wfc,
 		globalParams:  make(map[string]string),
+		volumes:       wf.Spec.DeepCopy().Volumes,
```

Comment:
Why make a new field when you can just access `woc.wf.Spec.Volumes`?

Reply:
This was my misunderstanding as well, but @jessesuen explained it to me (the conversation is lost due to my force pushes).

We need to copy this field (or any field we need to modify) because `woc.wf` is _always_ the value in the etcd. Thus, there is no reproducibility.
If we use `woc.wf.Spec.Volumes` and the user wants to let's say `retry` the wf, the `{{...}}` strings will be substituted the first time the wf was run, so the behavior will not be as desired.